### DISCUSSION TITLE
File manager redesign

### DISCUSSION THREAD

**Discussion by NicKoehler** [rustdesk/rustdesk#3288] [General]
It took some time, but I redesigned the file manager (it's only a mockup). I don't even know if I'll be able to implement it, but before I even try, I need to know if this meets your preferences. What do you think?

**Light theme**

![file manager light](https://user-images.githubusercontent.com/53040044/220041170-2eae4959-0283-488d-a194-330e4f180e28.png)

**Dark theme**

![file manager dark](https://user-images.githubusercontent.com/53040044/220041233-a3f7e9b2-4a1f-45be-80da-a583751236ba.png)

**Comment by rustdesk** [comment]
I love your design, it meets my preference very much.

**Comment by rustdesk** [comment]
I only have one concern, the right arrow on the rightmost side may cause confusion, because usually users will expect to click here and go to next page. But our arrow only refers to the transfer direction.

![image](https://user-images.githubusercontent.com/71636191/220045971-8c7e2025-4486-44eb-9bd9-8c8fdfd7c0dc.png)

**Reply by NicKoehler** [reply]
That's what I thought. The arrow represents the transfer direction. Every button has a rounded square around the icon, but that arrow doesn't have one. Do you think it's still confusing?

**Reply by rustdesk** [reply]
Yes, still confusing for me, because it is at the right most side.

**Reply by rustdesk** [reply]
How about put it at the left side of `Speed`?

**Reply by NicKoehler** [reply]
Something like this?

![immagine](https://user-images.githubusercontent.com/53040044/220069322-5c502e0d-9b5b-451c-ae70-f7c4c5be1822.png)

**Reply by rustdesk** [reply]
How about like this?

![image](https://user-images.githubusercontent.com/71636191/220070802-b2820241-3388-44af-8979-ae91517f3777.png)

**Reply by NicKoehler** [reply]
![immagine](https://user-images.githubusercontent.com/53040044/220072074-af14fcc9-e040-400d-aa97-f0582d09246d.png)

**Reply by rustdesk** [reply]
Yes, space is used more efficiently.

**Comment by NicKoehler** [comment]
WIP

![image](https://user-images.githubusercontent.com/53040044/220410604-5b48d3e7-1d1e-4847-91db-19d448c2e226.png)

**Reply by rustdesk** [reply]
good job

**Comment by grummbeer** [comment]
Very nice! 

is it a good idea share the look with home/settings page? Bringing also the [title bar](https://github.com/rustdesk/rustdesk/pull/3283#issuecomment-1439352585) in harmony with this new look?

would help on it.

**Reply by NicKoehler** [reply]
Probably it's the next thing to do

**Reply by grummbeer** [reply]
I am currently working on the Peers component and would adopt the same look for it.

- Size/spacing
- Icons
 
![file-transfer-share-look](https://user-images.githubusercontent.com/67791701/220866010-efa04715-1e5c-45ad-ba5c-09c068ed4e46.png)


Is it right to use the new icons instead of the material icons, even if they look the same? 

see "three dots" and "search".

- Icons.more_vert >  dots.svg
- Icons.search > search.svg

**Reply by grummbeer** [reply]
> Is it right to use the new icons instead of the material icons, even if they look the same?

I think its consequent, so yes?

**Reply by NicKoehler** [reply]
They look the same, but I think it's better to use a custom SVG so that if I want to change it, it will automatically update for the entire application

**Comment by bovirus** [comment]
@rustdesk 

Please consider when you can to redisign the defult column width in File Manager

"Modified" and "Size" have a too big width.
"Size" has a big size and cannot be reduced as expecyted.
For this reason the "Name" column width cannot be increase moe then a because "Size" cannot be reduced.

**Comment by rustdesk** [comment]
You can easy to modify it, then build it via github ci until reach your requirement. Then if it is ok, you can submit the PR. If you are waiting for us, you may wait for long, because we have a very big waitlist.

**Comment by bovirus** [comment]
@rusdesk

Could help to me to define the default width for columns?
What's the file thatd efine the default column width in File Manager? 
Thanks. KR.

**Comment by rustdesk** [comment]
Search the keywords in the source code, you will find. Frankly, I do not know either, this part of code is not written by me.

**Comment by bovirus** [comment]
@rustdesk 

I hope that someone can help me.
I hope the aiuhtor of these part of the code.

It's  a very boring situation.
If you have a longer file name the full name is truncated.
If you have same files with longer name and starting part of the name is equal it's diffuclt to check the filename.

**Reply by rustdesk** [reply]
please @ the author, you can find him in this discussion.

**Comment by bovirus** [comment]
@NicKoehler

Could you please check and apply changes to fix the issue (columns size)?

**Reply by NicKoehler** [reply]
i'll take a look when I have some free time

**Reply by NicKoehler** [reply]
I'm working to this type of behaviour:

- when a column is resized other columns are not moving
- when the entire window is resized the columns shrinks too

What do you guys think?

https://github.com/rustdesk/rustdesk/assets/53040044/026a502f-5a96-4da9-aa20-0a73d4c23206

**Reply by bovirus** [reply]
@NicKoehler

Yes is this,
Currently you cannot resize "Size" column width under a value that is also too big for "Size",
My scope is to maximize column width for "file name" leaving just the right space for dat and siize.
Another option is to save (and restore at next run) column width settings.

**Comment by Wolf-SO** [comment]
Hello everyone. <br> I really like the GUI for file transfer. Thank you very much for the good work you have done here. <br>What I particularly like is that the last locations (directory) on both sides are saved for the next session.

Unfortunately, there is a usability limitation that I found concerning directory management: <br><sup>(I'm working on Windows 10 Pro 22H2 if this matters.)</sup>

It's quite difficult to get or set the current directory on each page via copy & paste when the paths get longer: you have to click exactly on one of the **`>`** symbols to not accidentally change the directory. (BTW, the click on the last portion of the path does not have any effect.)

I also noticed that after such a `>`-click, the focus of the display remains on the beginning of the path (from the root) and not on the usually much more specific end, on which the display always stands when restarting.

In my opinion, the usability would improve by letting the click of the last part enter plain mode, as well as that the default presentation would truncate long paths on the root side.

I wasn't sure if I should file a bug report for this, so I'll use *Discussion* first to see what you think of it. Thanks.
